configuration/azure/appconfig/appconfig.go

```diff
@@ -248,9 +262,75 @@ func (r *ConfigurationStore) getLabelFromMetadata(metadata map[string]string) *s
 }
 
 func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
-	return "", fmt.Errorf("Subscribe is not implemented by this configuration store")
+	sentinelKey := r.getSentinelKeyFromMetadata(req.Metadata)
+	if sentinelKey == "" {
+		return "", fmt.Errorf("sentinel key is not provided in metadata")
+	}
+	uuid, err := uuid.NewRandom()
+	if err != nil {
+		return "", fmt.Errorf("failed to generate subscription id, error is %w", err)
+	}
+	subscribeID := uuid.String()
+	stop := make(chan struct{})
```

Comment:
Rather than using a separate channel, create a context that is a child of `ctx` and store that in `subscribeStopChanMap`. To unsubscribe, then, cancel that context.